pennylane/vqe/vqe.py

```diff
@@ -129,7 +125,10 @@ def __init__(self, coeffs, observables, simplify=False, group=False, id=None):
         if group:
             self.group()
 
-        super().__init__(coeffs, wires=self._wires, id=id, do_queue=False)
+        coeffs_flat = [coeffs[i] for i in range(qml.math.shape(coeffs)[0])]
+        # overwrite, now that we have the info
+        self.num_params = len(coeffs_flat)
```

Comment:
Hack: overwrite the static `num_params` here, since we know how many coefficients we have